src/base/NekRSSeparateDomainProblem.C

```diff
@@ -336,4 +432,33 @@ NekRSSeparateDomainProblem::temperature(const int elem_id, const double temperat
   }
 }
 
+void
+NekRSSeparateDomainProblem::scalar(const int elem_id, const int scalarId, const double scalar)
+{
+  const auto & bc = _nek_mesh->boundaryCoupling();
+
+  // We can only write into the nekRS scratch space if that face is "owned" by the current process
+  if (nekrs::commRank() == bc.processor_id(elem_id))
+  {
+    nrs_t * nrs = (nrs_t *)nekrs::nrsPtr();
+    mesh_t * mesh = nekrs::entireMesh();
+
+    int scalarFieldOffset = nekrs::scalarFieldOffset();
+    int scalarWrkOffset = (scalarId + 1)*scalarFieldOffset; // offset by 1, first "scalar" is temperature
```

Comment:
I think this should be indexed like `scalarId * scalarFieldOffset`, right? Because we now have `scalarID >= 1`.

Reply:
My thought process is that for this coupling class, we'd keep the following portions of `usrwrk` always reserved for these following coupled values. 

`usrwrk[id]` reserved for Velocity
`usrwrk[id + scalarFieldOffset]` reserved for Temperature, even if temperature isn't solved for
`usrwrk[id + 2*scalarFieldOffset]` reserved for Scalar01, even if it isn't solved for
`usrwrk[id + 3*scalarFieldOffset]` reserved for Scalar02, even if it isn't solved for
`usrwrk[id + 4*scalarFieldOffset]` reserved for Scalar03, even if it isn't solved for

So `scalarWrkOffset = (scalarId + 1)*scalarFieldOffset` for the scalars because `usrwrk[id + scalarFieldOffset]` is reserved for temperature no matter what.

Reply:
I see your point - if in the future we wanted to couple your NekRS domain to a solid heat conduction model, then we'd still need the 0th and 1st slots in `usrwrk` to hold heat flux and temperature, respectively. 

We'll just need to be sure this is well documented so people don't get confused with the indexing.